src/EventListener/ExceptionListener.php

```
@@ -0,0 +1,307 @@
+            array_keys(self::$exceptionTemplates)
+        );
+        if (!empty($candidates)) {
+            $template = self::$exceptionTemplates[array_shift($candidates)];
+            return $this->createTemplateResponseFromException($template, $exception);
+        }
+
+        // Unknown HttpExceptionInterface implementing exception, no way to handle.
+        return null;
+    }
+
+    /**
+     * Try to create a response for the given template.
+     *
+     * @param string                 $template  The name of the template to render.
+     * @param HttpExceptionInterface $exception The exception to render.
+     *
+     * @return Response
+     */
+    private function createTemplateResponseFromException($template, HttpExceptionInterface $exception)
+    {
+        return new Response(
+            $this->renderErrorTemplate($template),
+            $exception->getStatusCode(),
+            $exception->getHeaders()
+        );
+    }
+
+    /**
+     * Try to render an error template.
+     *
+     * @param string $template The template name. Will get searched at standard locations.
+     *
+     * @return string
+     */
+    private function renderErrorTemplate($template)
+    {
+        // TODO: make twig templates out of these.
+        if ($response = $this->tryReadTemplate(sprintf('%s/templates/%s.html5', $this->rootDir, $template))) {
+            return $response;
+        }
+
+        return $this->tryReadTemplate(
+            sprintf('%s/../Resources/contao/templates/backend/%s.html5', __DIR__, $template)
+        );
+    }
+
+    /**
+     * Try to read a template.
+     *
+     * @param string $template The file path to test.
+     *
+     * @return null|string
+     */
+    private function tryReadTemplate($template)
+    {
+        if (file_exists($template)) {
+            // Isolate the template parsing, the "unused" root dir variable will get used in the template.
```

@aschempp For symfony this works but we are in Contao and swallow errors and therefore I guess we will have a warning `assuming $this from incompatible context` and you will still have `$this` available.
Have not checked though.